### PR TITLE
Arnej/hnsw search preview.

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst @havardpe first draft, please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp [hunk 1]
```diff
@@ -310,6 +310,43 @@ HnswIndex::remove_document(uint32_t docid)
     _node_refs[docid].store_release(invalid);
 }
 
+std::vector<uint32_t>
+HnswIndex::find_top_k(TypedCells vector, uint32_t k)
+{
+    std::vector<uint32_t> result;
+    std::vector<HnswCandidate> candidates = top_k_candidates(vector, k + 100);
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp]: Perhaps 'k + 100' should be a parameter to the function instead, e.g. 'search_k'?

**Review by geirst — APPROVED** [review]
Overall looks good, please consider the comment.

File: searchlib/src/tests/tensor/hnsw_index/hnsw_index_test.cpp [hunk 2]
```diff
@@ -86,6 +86,24 @@ class HnswIndexTest : public ::testing::Test {
         ASSERT_EQ(exp_levels.size(), act_node.size());
         EXPECT_EQ(exp_levels, act_node.levels());
     }
+    void expect_top_3(uint32_t docid, std::vector<uint32_t> exp_hits) {
+        uint32_t k = 3;
+        auto qv = vectors.get_vector(docid);
+        auto rv = index->top_k_candidates(qv, k);
+        size_t idx = 0;
+        for (const auto & hit : rv) {
+            // fprintf(stderr, "found docid %u dist %.1f\n", hit.docid, hit.distance);
```
Comment (geirst) [inline searchlib/src/tests/tensor/hnsw_index/hnsw_index_test.cpp]: Consider removing this.

File: searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp (hunk 1 above)
Comment (arnej27959) [reply searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp]: now exposed as "explore_k"

**Comment by arnej27959** [comment]
@geirst PTAL

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**